addons/website_form/static/src/snippets/s_website_form/options.js

```diff
@@ -1,484 +1,711 @@
-                    field.name = field_name;
-                });
-                return fields;
+            field.formatInfo = {
+                labelPosition: this._getlabelPosition(),
+                labelWidth: this.$target[0].querySelector('.o_we_form_label').style.width,
+            };
+            field.col = Array.from(classList).filter(el => el.match(/^col-/g)).join(' ');
+        },
+        /**
+         * @private
+         * @param {Object} field
+         * @returns {Promise<HTMLElement>}
+         */
+        _renderField: function (field) {
+            return this._getFieldRecords(field).then(() => {
+                const template = document.createElement('template');
+                template.innerHTML = qweb.render("website_form.field_" + field.type, {field: field}).trim();
+                const html = template.content.firstChild;
+                html.dataset.name = "Field";
+                return html;
             });
         },
+        /**
+         * @private
+         * @returns {string}
+         */
+        _getlabelPosition: function () {
+            const label = this.$target[0].querySelector('.o_we_form_label');
+            if (this.$target[0].querySelector('.row')) {
+                if (label.classList.contains('text-right')) {
+                    return 'right';
+                } else {
+                    return 'left';
+                }
+            } else {
+                if (label.classList.contains('d-none')) {
+                    return 'none';
+                } else {
+                    return 'top';
+                }
+            }
+        },
+        /**
+         * @private
+         * @returns {string}
+         */
+        _getPlaceholder: function () {
+            const input = this._getPlaceholderInput();
+            return input ? input.placeholder : '';
+        },
+        /**
+         * Returns the field's input if it is placeholder compatible, else null
+         *
+         * @private
+         * @returns {HTMLElement}
+         */
+        _getPlaceholderInput: function () {
+            return this.$target[0].querySelector('input[type="text"], input[type="email"], input[type="number"] , textarea');
+        },
+        /**
+         * Set the placeholder on the current field if the input allow it
+         *
+         * @private
+         * @param {string} value
+         */
+        _setPlaceholder: function (value) {
+            const input = this._getPlaceholderInput();
+            if (input) {
+                input.placeholder = value;
+            }
+        },
+        /**
+         * Returns the type of the  field, can be used for both custom and existing fields
+         *
+         * @private
+         * @returns {string}
+         */
+        _getFieldType: function () {
+            return this.$target[0].dataset.type || this.$target[0].querySelector('.o_we_form_label').getAttribute('for');
```

Comment:
Use DOM properties instead of attributes when you can. So here, use `.for`

Reply:
I think it is `.htmlFor`

https://developer.mozilla.org/en-US/docs/Web/API/HTMLLabelElement/htmlFor

Reply:
:+1: